### DISCUSSION TITLE
x-model.fill with custom elements

### DISCUSSION THREAD

**Discussion by pelme** [alpinejs/alpine#3813] [7. Bugs (archived)]
Using x-model.fill with custom elements does not seem to work properly. It seems that Alpine does run too early and does not pick up .value as defined in the component.

See this example:
https://codepen.io/pelme/pen/gOZJJMp?editors=1010

Here - https://github.com/alpinejs/alpine/blob/8adfcec29122629eedaca50e9a1904675406411a/packages/alpinejs/src/directives/x-model.js#L73C1-L77C6 - the input event is directly dispatched. Maybe https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined could be used if .fill is used on a custom element before dispatching the input event.

**Comment by ekwoka** [comment]
Define your elements before starting alpine.

**Reply by pelme** [reply]
Hey @ekwoka, thanks for the suggestion. That's a possible workaround! The current behavior is a bit surprising though since everything with x-model / other Alpine stuff just works with custom elements. It is just `.fill` that does not work.

It would be nice it was handled automatically to not have to think about it. Alpine goes to great lengths to deal with new DOM nodes added at various points in the life cycle of a web page to "just work" which is amazing.

I think something like this could make it work:
```js
if (el.tagName.includes("-") {
    customElements.whenDefined(el.tagName).then(() => el.dispatchEvent(new Event(event, {})));
} else {
    el.dispatchEvent(new Event(event, {}))
}
```

Would you be *against* a change like this? If this contribution would be welcome I could try to create a PR for it!

**Reply by ekwoka** [reply]
I'm not a maintainer so go ahead :)
